lib/elixir/lib/protocol.ex

```diff
@@ -474,6 +474,8 @@ defmodule Protocol do
   defp list_dir(path), do: list_dir(to_charlist(path))
 
   defp extract_from_file(path, file, prefix, callback) do
+    path = to_charlist(path)
```

Comment:
Given that we also convert `to_charlist` inside `list_dir`, I would rather change the comprehension a couple lines above to this:

```
for path <- paths,
    path = to_charlist(path),
    file <- ...
```

Reply:
And then remove the `to_charlist` call on both `list_dir` and `extract_from_file`.

Reply:
Done